### DISCUSSION TITLE
Extending keystone-6/auth sessionData

### DISCUSSION THREAD

**Discussion by noor-codes** [keystonejs/keystone#7312] [Questions] [answered]
### Explanation
In KeystoneJS, the easiest and fun way to implement auth globally in the entire app is to inject data into the session in `auth.ts` file as highlighted down below where I can have access to id, name, email and role of the user with all the permissions.

```diff
const { withAuth } = createAuth({
  listKey: 'User',
  identityField: 'email',

+  sessionData: `id name email   
+ role {
+     ${permissionsList.join(' ')}
+      name
+    }
+  `,

  secretField: 'password',
  initFirstItem: {
    fields: ['name', 'email', 'password']
  }
})
```

### Problem
1: The `sessionData` field is mapped to querying the data on the `User` table in my case.
2: If I want to access data that isn't in the User table. I need to make relationship with that field to be able to query it and stick it in the session.

```graphql
query {
  user(where: {id: "cl077jpxk0079c8gppukswjx5"}) {
    id
    name
    username
    email
    role {
      id
      name
    }
  }
}
```

### What I want
- I want to be able to stick some data from a table that has no relationship with the user table, like Global Settings Table.
- Keystone should let me put anything in the session, right now I am limited to users table only on the condition to include other tables if there is a relationship between User and that table. 
- In my case, I have `Control` table that has no relationship with other tables and I want the `dark_mode` field to be in the session.
- I also added checks to make sure there will be only one row here.

![image](https://user-images.githubusercontent.com/28804764/156068612-ccc7b498-3aa4-47d5-adc5-aac55a7913a7.png)

I'd be glad if anyone shows me a way around it.
Thanks.

**Comment by emmatown** [comment, marked as answer]
You can do this by augmenting the session config like this:

```ts
const ksConfig = withAuth(
  config({
    // your keystone config
  })
);

export default config({
  ...ksConfig,
  session: {
    ...ksConfig.session!,
    async get(args) {
      const control = await args
        .createContext({ sudo: true })
        .query.Control.findOne({ where: { id: 'whatever' } });
      return {
        ...ksConfig.session!.get(args),
        control,
      };
    },
  },
});

```

**Reply by noor-codes** [reply]
Wow! Thank You so much!

**Reply by noor-codes** [reply]
I'll give it a try.
